### PR TITLE
BAK-1207 Create bulk import endpoint

### PR DESCRIPTION
**Pull request by joel-wright**

### CODE REVIEW

File: backend/app/api/api_v1/routers/admin.py [hunk 1]
```diff
@@ -31,7 +53,7 @@
 
 @r.get(
     "/users",
-    response_model=List[User],
+    response_model=list[User],
```
Comment (diversemix) [inline backend/app/api/api_v1/routers/admin.py:56]: Ooo good spot

File: backend/app/api/api_v1/schemas/document.py [hunk 2]
```diff
@@ -127,4 +128,17 @@ class RelationshipAndDocumentsGetResponse(BaseModel):
...
+
+
+class DocumentUploadCompleteRequest(BaseModel):
+    """Information generated during the upload of a document that should be stored."""
+
+    md5_sum: str
+    content_type: str
+
+
+class BulkImportResult(BaseModel):
```
Comment (diversemix) [inline backend/app/api/api_v1/schemas/document.py:141]: Without going to full on Java ... I was thinking of naming this `BulkImportValidatedResult` -  to signify that the request has been successfully validated as well as initiated?

File: backend/app/core/validation/util.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+from app.api.api_v1.routers.lookups.util import get_metadata
+
+_LOGGER = logging.getLogger(__file__)
+
+
+def _flatten_maybe_tree(
+    maybe_tree: Sequence[Mapping[str, Any]],
+    values: Optional[set[str]] = None,
+) -> Collection[str]:
```
Comment (diversemix) [inline backend/app/core/validation/util.py:14]: I know you said you would ... but putting a note here for unit tests :D

**Review by diversemix — APPROVED** [review]
lgtm

File: backend/app/core/validation/util.py (hunk 3 above)
Comment (joel-wright) [reply backend/app/core/validation/util.py:14]: done

File: backend/app/api/api_v1/schemas/document.py (hunk 2 above)
Comment (joel-wright) [reply backend/app/api/api_v1/schemas/document.py:141]: agreed, we only return this on success, so this is a validated count now (and maybe more later)

Comment (diversemix) [reply backend/app/api/api_v1/schemas/document.py]: Great!

**Pull request merged**